File: Python/bubble_sort.py
import random


def bubble_sort(arr):
    length = len(arr)

    for index in range(length):
        swapped = False
        unsorted = length - index - 1
        
        for j in range(unsorted):
            if arr[j]>arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = True

        if not swapped:
            break


if __name__ == '__main__':
    random.seed(0)

    arr = [random.randint(-100,100) for _ in range(10)]
    print('Array: ', arr)

    bubble_sort(arr)
    print('Sorted array: ', arr)

File: C++/bubble_sort.cc
#include <iostream>
#include <vector>


void BubbleSort(std::vector<int>& arr){
    int length = arr.size();

    for(int i=0; i<length; i++){
        bool swapped = false;
        int unsorted = length - i - 1;

        for(int j=0; j<unsorted; j++){
            if(arr[j]>arr[j+1]){
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;

                swapped = true;
            }
        }

        if(!swapped)
            break;
    }
}


void Print(const std::vector<int>& arr){
    for(const auto& entry : arr){
        std::cout << entry << " ";
    }

    std::cout << std::endl;
}


int main(){
    std::vector<int> arr = {-2, 94, 7, -90, -34, 30, 24, 3, 100, -23};

    Print(arr);
    BubbleSort(arr);
    Print(arr);
}